android/java/org/chromium/chrome/browser/crypto_wallet/activities/BraveWalletBaseActivity.java

```diff
@@ -48,6 +54,16 @@ public void onUserInteraction() {
 
     @Override
     public void onConnectionError(MojoException e) {
+        if (mKeyringServiceObserver != null) {
+            mKeyringServiceObserver.close();
+            mKeyringServiceObserver.destroy();
+            mKeyringServiceObserver = null;
+        }
+        if (mTxServiceObserver != null) {
+            mTxServiceObserver.close();
+            mTxServiceObserver.destroy();
+            mTxServiceObserver = null;
```

Comment:
fix this and similar places after https://github.com/brave/brave-core/pull/14725#discussion_r951172290 fix

Reply:
they are intended to be different, just not currently in that PR, but they will be in the future when a problem with closing observer connection is solved.